src/sqs/error-handlers.ts

```diff
@@ -86,9 +87,12 @@ const maybeHandleNonRetryableResponseCode = <MessagePayload extends BaseMessageP
 const maybeHandleRateLimitingError = <MessagePayload extends BaseMessagePayload>(error: Error, context: SQSMessageContext<MessagePayload>): ErrorHandlingResult | undefined => {
 	if (error instanceof GithubClientRateLimitingError) {
 		context.log.warn({ error }, `Rate limiting error, retrying`);
-		const delaySec = error.rateLimitReset + RATE_LIMITING_DELAY_BUFFER_SEC - (Date.now() / 1000);
-		return { retryable: true, retryDelaySec: delaySec, isFailure: true };
+		const buffer = BASE_RATE_LIMITING_DELAY_BUFFER_SEC - context.receiveCount * 10;
+		const rateLimitReset = error.rateLimitReset + buffer - Date.now() / 1000;
+		const retryDelaySec = rateLimitReset + ONE_HOUR_IN_SECONDS * (context.receiveCount - 1);
```

Comment:
That's quite aggressive. I think we need to put a comment clarifying this also

Reply:
Commenting.